setup.py

```diff
@@ -0,0 +1,134 @@
+#
+#   Copyright 2013-2022 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+import inspect
+import os
+import subprocess
+import shlex
+
+from setuptools import (
+    setup,
+    Extension,
+    find_packages,
+)
+
+import setuptools.command.build_ext
+
+
+class build_ext(setuptools.command.build_ext.build_ext):
+    """
+    Custom setuptools command class to puppet CMake.
+    """
+
+    def build_extension(self, _ext: Extension):
+        """
+        Hook called by setuptools to build the given Extension.
+
+        @param _ext: Extension to build (we only have one, so ignored).
+        """
+        self.__cmake(
+            [
+                "-S",
+                ".",
+                "-B",
+                self.build_temp,
+                "-G",
+                "Ninja",
+                # Place output artifacts where setuptools expects.
+                "--install-prefix",
+                os.path.abspath(self.build_lib),
+                "--preset",
+                "setuptools",
+            ]
+        )
+
+        self.__cmake(["--build", self.build_temp, "--target", "openassetio-python-module"])
+
+        self.__cmake(
+            [
+                "--install",
+                self.build_temp,
+                "--component",
+                "openassetio-python-module",
+            ]
+        )
+
+    def __cmake(self, args: list[str]):
+        """
+        Execute `cmake` as a subprocess with given arguments.
+
+        @param args: Command-line arguments to pass to `cmake`.
+        """
+        args = ["cmake"] + args
+        self.announce(" ".join(map(shlex.quote, args)), level=2)
+        subprocess.check_call(args, env=os.environ.copy(), stderr=subprocess.STDOUT)
+
+
+setup(
+    name="openassetio",
+    version="1.0.0a4",
+    description=(
```

Comment:
[quesion] On [further reading](https://packaging.python.org/en/latest/tutorials/packaging-projects/#creating-pyproject-toml) I wonder if we should bundle as much metadata as possible in `pyproject.toml` to do things the "modern" way?

Reply:
It would indeed be a good idea.

Reply:
Moved all the metadata to `pyproject.toml` in e9807bcd